apps/src/lib/kits/maker/CircuitPlaygroundBoard.js

```diff
@@ -130,6 +130,8 @@ export default class CircuitPlaygroundBoard extends EventEmitter {
       );
     }
 
+    this.fiveBoard_.on('close', () => this.emit('disconnect'));
+
     this.fiveBoard_.on('disconnect', () => this.emit('disconnect'));
```

Comment:
What's the difference between a `close` event and a `disconnect` event?

Reply:
I'm not 100% sure about any of this, but from looking at the ChromeSerialBridge logs while manually testing, disconnect events happen when you unplug or press the reset button. Close events happen when another connection to the board is made (the case I was testing was connect to the board in tab 1, then connect in tab 2, should disconnect tab 1). 
https://github.com/code-dot-org/chrome-serial-app/blob/master/app/scripts/ChromeSerialBridge.js#L54 
https://github.com/code-dot-org/chrome-serial-app/blob/master/app/scripts/ChromeSerialBridge.js#L69
johnny-five listens for these events and re-emits them: https://github.com/code-dot-org/johnny-five/blob/master/lib/board.js#L345

So (again not 100% sure about really anything related to circuit playgrounds), I think by just listening for the close event and treating it as a disconnect, we can handle the two tab use-case. From manual testing, it seems to work: click run in tab 1 (makes connection to the board), switch to tab 2 and click run (makes connection to the board, causes tab 1 to lose its connection, which triggers applab reset), switch back to tab 1 and click run (same thing- makes connection to the board, which causes tab 2 to lose connection and reset).

Reply:
Awesome, I forgot to consider the multi-tab case.  This sounds great.

Reply:
Turns out this doesn't work. Since both tabs receive the close event, they both disconnect (not just the one that was already running).
Tab 1 connects/runs. Tab 2 clicks run. This issues a close event. Tab 1 disconnects, but also Tab 2 disconnects.
I'll need to figure out a better fix for this, but probably won't get to it until next week.